### ISSUE TITLE
Browsing with cookies in header does not work in iOS

### ISSUE THREAD

**Issue by mathiasgmz** [react-native-webview/react-native-webview#3535]
**Bug description:**

Hello! I have tried to add a session cookie in the request header from the new version (13.10.6) of the webview with the parameters sharedCookiesEnabled={true} and thirdPartyCookiesEnabled={true}. And it keeps returning me to the login screen. Any ideas? How can I configure the webview so that it can be navigated freely with the session cookie set in the header?
Currently I have it like this:

const dataReq = { uri: https://mydomain/${currentURI}`,
headers: {
Cookie: cookie, //cookie extracted from CookieManager
}}`

<WebView sharedCookiesEnabled={true} thirdPartyCookiesEnabled={true} source={dataReq} />

Any ideas?

**Expected behavior:**

The navigation should work the same as Android, that is, when putting the session cookies in the header, it should go directly to the page I need, it should not redirect me to the login.

**Environment:**
- OS: iOS
 - react-native version: 0.74.1
 - react-native-webview version: 13.10.6

**Comment by smohadjer** [comment]
Hi @mathiasgmz Did you find a solution? I ran into the same problem today!

**Comment by mathiasgmz** [comment]
Hello @smohadjer! Hi!, I haven't been able to solve it completely, so what I've done is:
- As "source" I've put the login url, this url returns the cookie session in the header.
- I've programmed an "onLoad" event for to spinner disappears.
- I've programmed an "onNavigationStateChange" event like this:
`if (url.includes(<page to linked when you log in>)) {
const newURL = `<domain>/${currentURI}`; // endpoint that I want to access
const redirectTo = 'window.location = "' + newURL + '"';
webViewRef.current.injectJavaScript(redirectTo);
}`
- This way, when the component is executed, first, it must to login, then, the cookie session is obtained in the webview and it save it, and then it is redirected to the final URL that I want to access, it has only worked for me in this way.  While all this is happening, the spinner is showing, and it seems like the endpoint is being accessed. Sorry for my bad english :)
